### ISSUE TITLE
More sensible prefix + unit combinations

### ISSUE THREAD

**Issue by cheezman34** [hgrecco/pint#273]
As it is now, pint is pretty liberal about combining prefixes + units.  For example, "pinch" is parsed as a "picoinch", which doesn't seem very useful to me.  Does anyone have any thoughts on supporting more fine-grained prefix + unit name combinations?  It would be helpful when parsing unverified input to have more restrictive unit names.  We could allow prefix combinations to be specified in the definitions file, so people could write their own rules.

Possible rules:
1) SI prefixes only with SI units (no "pinch" like above)
2) binary prefixes only with information units ("kibibyte" is useful, "kibimeter" less so)
3) abbreviated prefixes only with abbreviated units (allow "km" but no "kmeter")
4) no prefixes when a unit name has one built in (disallow "kilokph")

Comments?  Dissenting opinions?

**Comment by hgrecco** [comment]
Something like this was discussed in the past and the agreement was: it would be nice but is not a priority. After all, we are consenting adults. Now that pint has moved forward a lot, we could thing of adding something  like this  after #215 is well tested and in use as long as it does not complicate the code too much.

**Comment by jdreaver** [comment]
I think this would add too much complexity to pint. Right now, the rules for adding SI prefixes are very simple. Adding this feature would require:

* Adding the ability to categorize prefixes and units into groups (SI, not SI, binary, etc).
* Specifying which names are abbreviations (like "ft" for feet), and which are just plural or synonyms (like foot).
* Specifying that a unit has a builtin prefix (like in your kph example).

I think that @hgrecco is correct when he says that "we are all consenting adults," which is also part of the philosophy of Python.

If this could be implemented very simply, and was somehow optional, then I think it would be very neat. I just don't see a way to do it that easily.

**Comment by hgrecco** [comment]
I am closing this. Feel free to reopen.

**Comment by OJFord** [comment]
This would also help in a hypothetical `to_unprefixed_unit` method, which I was going to request until I found this issue.

(`to_root_unit` will take both `kilograms` and `metric_tons` to `grams`, which isn't the same as just stripping the `kilo` SI prefix from `kilograms`.)
